lib/add-nested-value.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+const addNestedValue = function (pojo, name, value) {
+  const recurse = function recurse(pojo, keys, value) {
```

Comment:
This is much, much clearer now.

I was making several conceptual errors, one of which was thinking that `const recurse = ...` was what made the code above a function expression. That's not correct. The fact that `function` is not the first word in the statement makes the expression on the RHS of assignment a "function expression".

Additionally, the name for the function expression is not `const recurse`, it's the name given before the argument list in the function expression.

The reason to use named function expressions over function expressions is to enable recursion.